### PR TITLE
Trigger PRs on `pull_request` trigger

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

**Comment by mgeplf** [comment]
what happens if you enble actions on your fork?

**Comment by wizmer** [comment]
They are already allowed.

**Comment by mgeplf** [comment]
They were allowed when you made a push?

**Comment by wizmer** [comment]
Yes. You mean in settings -> Actions, right ? Yes, they were.
I don't understand why the checks are still listed up there even though they are not being triggered.

**Comment by mgeplf** [comment]
> I don't understand why the checks are still listed up there even though they are not being triggered.

My only guess is b/c they are listed as `Required`?

**Comment by wizmer** [comment]
Ok, and now that I have changed the trigger `docstring_check` is being triggered.

File: .github/workflows/docstring_check.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: docstring-check
+
 on:
-    push:
+    pull_request:
```
Comment (mgeplf) [inline .github/workflows/docstring_check.yaml:4]: I guess it could also be `[pull_request, push]` for the best of both worlds?

Comment (wizmer) [reply .github/workflows/docstring_check.yaml:4]: I'm not sure keeping push adds value but it's debatable I guess. Also the jobs that list both pull_request and push will get triggered twice if you make a PR from a branch on BlueBrain remote.

**Comment by mgeplf** [comment]
> I'm not sure keeping push adds value but it's debatable I guess. 

I use it for WIP branches, before I make the PR, such that the PR doesn't fail testing.

**Comment by wizmer** [comment]
I see, I see, thanks for explaining.

**Comment by mgeplf** [comment]
any idea why coverage isn't working?

Also, will there be any interaction w/ the release process, do you think?

**Comment by wizmer** [comment]
Because I kept the coverage just on push since it is not being tested but it is rather informational.
And no that should not change anything.

**Comment by tomdele** [comment]
And we don t have the problem we had with the double run of the CI at every push or merger (I cannot remember...) ?

**Comment by wizmer** [comment]
We will have it now I think but only for PRs made on a BlueBrain branch

**Comment by wizmer** [comment]
I think I need to merge this because it seems there is a bug with coverage-test still expected even when I remove completely the file.

**Comment by mgeplf** [comment]
> I think I need to merge this because it seems there is a bug with coverage-test still expected even when I remove completely the file.

Could you wait until you have other's reviews before merging?

> Only trigger coverage on master push

Why only on master? I explained above how it's useful to see it on other branches as well.

**Comment by wizmer** [comment]
> Could you wait until you have other's reviews before merging?

I was trying to understand why coverage is not working as intended and needed the merge for this.

**Comment by wizmer** [comment]
> Why only on master? I explained above how it's useful to see it on other branches as well.

Where ? You explained that it is useful having the CI running because you can see where the tests are failing. This is not related to coverage. But feel free to change the behavior anyway, I was just trying to help.

**Comment by mgeplf** [comment]
> I was trying to understand why coverage is not working as intended and needed the merge for this.

I still think it's helpful to let others have a chance to review and comment.

> This is not related to coverage. 

I consider coverage as part of CI (like the docstring tests, and formatting); if it's going down, I want to know what I'm missing - sorry I should have been more clear.

**Comment by wizmer** [comment]
FYI: The coverage is still showing as expected because of the repository settings.

Daniel explained that to me this morning for an unrelated question: https://bbpteam.epfl.ch/project/issues/browse/HELP-13549

**Comment by mgeplf** [comment]
I don't follow; the thing that changed w/ coverage is that it's only on the master branch, so I would expect it to still show up and run on master pushes, and not run on non-master branches.

Are you referring to https://github.com/BlueBrain/libsonata/pull/129#issuecomment-772346837 ?

**Comment by wizmer** [comment]
As of today, if a PR is made, it won't trigger the coverage checks but those checks are still required for the PR to be "green". This is an inconsstency and I was wondering why. As for NeuroR, I was wondering why Travis was still listed in the list of required checks. The issue is in fact not related to this PR. (in fact before this PR the issue was even more present since the CI checks were required but there was no trigger). 

So one need sync the triggers of the PR and the required items in the repo options. Either coverage is required on PRs and we need to fix the github action, or it is not required and we need to fix the repo settings.

**Comment by mgeplf** [comment]
> As of today, if a PR is made, it won't trigger the coverage checks but those checks are still required for the PR to be "green".

Yeah, and I can't merge without overriding the required checks, which is a regression.

Coverage is required; it also seems that now we're also running each job twice.

**Pull request merged**